extensions/eclipse-che-theia-plugin-remote/src/node/hosted-plugin-remote.ts

```diff
@@ -177,4 +192,50 @@ export class HostedPluginRemote {
         return [].concat.apply([], [...this.pluginsMetadata.values()]);
     }
 
+    /**
+     * Sends request to retreive plugin resource from its sidecar.
+     * Returns undefined if plugin doesn't run in sidecar or doesn't exist.
+     * @param pluginId id of the plugin for which resource should be retreived
+     * @param resourcePath relative path of the requested resource based on plugin root directory
+     */
+    public requestPluginResource(pluginId: string, resourcePath: string): Promise<Buffer | undefined> {
+        if (this.hasEndpoint(pluginId) && resourcePath) {
+            return new Promise<Buffer | undefined>((resolve, reject) => {
+                const endpoint = this.hostedPluginMapping.getPluginsEndPoints().get(pluginId);
+                const targetWebsocket = this.endpointsSockets.get(endpoint);
+                if (!targetWebsocket) {
+                    reject(new Error(`No websocket connection for plugin: ${pluginId}`));
+                }
+
+                this.resourceRequests.set(this.getResourceRequestId(pluginId, resourcePath), resolve);
+                targetWebsocket.send(JSON.stringify({
+                    'internal': {
+                        'method': 'getResource',
+                        'pluginId': pluginId,
+                        'path': resourcePath
+                    }
+                }));
```

Comment:
Do we need to call `resolve()` somewhere in the promise?

Reply:
We have to call resolve, but not here. It is called [here](https://github.com/eclipse/che-theia/blob/75990857f320bebf9e511b2f13a72b64dbb75f04/extensions/eclipse-che-theia-plugin-remote/src/node/hosted-plugin-remote.ts#L233).